### PR TITLE
Refactor member input

### PR DESCRIPTION
**Pull request by ling-ui**
Changes:
- Refactored member input to take in member index based on member list instead of member’s name for the following commands:
1. add task
2. edit task
3. assign task

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddTaskCommand.java [hunk 1]
```diff
@@ -16,24 +29,71 @@ public class AddTaskCommand extends Command {
...
-            + "Example: " + COMMAND_WORD + " \"merge PR#12\"";
+            + "Parameters:"
+            + FLAG_NAME_STR + " NAME "
+            + FLAG_DEADLINE_STR + " DEADLINE "
+            + FLAG_ASSIGNEE_STR + " ASSIGNEE "
+            + "Example: " + COMMAND_WORD + " "
+            + FLAG_NAME_STR + " \"Review PR\" "
+            + FLAG_DEADLINE_STR + " \"2023-12-04 23:59\" "
+            + FLAG_ASSIGNEE_STR + " \"Alex Yeoh\" "
+            + FLAG_ASSIGNEE_STR + " \"Bernice Yu\" ";
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/AddTaskCommand.java:40]: wait can we take in the name without any flag? i.e. the default argument should be the name

File: src/main/java/seedu/address/logic/commands/EditTaskCommand.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+ */
+@CommandLine.Command(name = "task")
+public class EditTaskCommand extends Command {
+    public static final String COMMAND_WORD = "edit task";
+
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Edits a current task identified by the index number used in the displayed task list. \n"
+                    + "Existing values will be overwritten by the input values. \n"
+                    + "Parameters: INDEX (must be a positive integer) "
+                    + FLAG_NAME_STR + " NAME "
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/EditTaskCommand.java:39]: the name flag here is okay i guess

File: src/main/java/seedu/address/logic/commands/AddTaskCommand.java (hunk 1 above)
Comment (ling-ui) [reply src/main/java/seedu/address/logic/commands/AddTaskCommand.java]: Removed name flag

File: docs/UserGuide.md [hunk 3]
```diff
@@ -140,18 +140,31 @@ Format:
...
 - `list members`
 
-### Add task to team : `add_task`
+### Add task to team : `add task`
 
-Add a new task to the current team.
+Add a new task to the current team. The task name is compulsory, while the other fields are optional.
 
 Format:
+* `add task -n TASK_NAME -a ASSIGNEES -d DEADLINE`
```
Comment (Devanshshah1309) [inline docs/UserGuide.md:148]: Do you think we should include quotes to make it very clear that the user needs to input quotes too?

Also, can you include another example with adding multiple assignees simultaneously?
```suggestion
* `add task -n "TASK_NAME" -a "ASSIGNEES" -d "DEADLINE"`
```

File: src/main/java/seedu/address/model/team/Task.java [hunk 4]
```diff
@@ -100,7 +103,15 @@ public String getDeadlineAsString() {
         if (deadline == null) {
             return "";
         } else {
-            return String.format("(By %s)", deadline.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")));
+            return String.format(" (By %s)", deadline.format(DateTimeFormatter.ofPattern(DATE_FORMAT)));
```
Comment (Devanshshah1309) [inline src/main/java/seedu/address/model/team/Task.java:106]: See if this works better
```suggestion
            return String.format(" (By %s)", deadline.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm a")));
```

**Review by Devanshshah1309 — APPROVED** [review]
LGTM just minor changes required.

**Pull request merged**